tests/TestCase/ORM/EntityTest.php

```diff
@@ -1656,4 +1676,101 @@ public function testHasValue(): void
         $this->assertTrue($entity->hasValue('floatNonZero'));
         $this->assertFalse($entity->hasValue('null'));
     }
+
+    /**
+     * Test isOriginalField()
+     */
+    public function testIsOriginalField(): void
+    {
+        $entity = new Entity(['foo' => null]);
+        $return = $entity->isOriginalField('foo');
+        $this->assertSame(true, $return);
+
+        $entity = new Entity([]);
+        $entity->set('foo', null);
+        $return = $entity->isOriginalField('foo');
+        $this->assertSame(false, $return);
+
+        $return = $entity->isOriginalField('bar');
+        $this->assertSame(false, $return);
+    }
+
+    /**
+     * Test getOriginalFields()
+     */
+    public function testGetOriginalFields(): void
+    {
+        $entity = new Entity(['foo' => 'foo', 'bar' => 'bar']);
+        $entity->set('baz', 'baz');
+        $return = $entity->getOriginalFields();
+        $this->assertEqualsCanonicalizing(['foo', 'bar'], $return);
+        $this->assertNotEqualsCanonicalizing(['foo', 'bar', 'baz'], $return);
```

Comment:
Whats those asserts? if the first matches, I dont quite see the point of the 2nd opposite here.